### PR TITLE
Decorate a class

### PR DESCRIPTION
**Pull request by Mujeeb4582**
# Task completed
- Create Nameable Class
- Make the Person class inherits from Nameable
- Create base Decorator Class and inherits from Nameable
- Create CapitalizeDecorator Class and inherits from Decorator
- Create TrimmerDecorator Class and inherits from Decorator

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -7,17 +13,52 @@ def initialize(age, name = 'Unknown', parent_permission: true)
     @name = name
     @age = age
     @parent_permission = parent_permission
+    super
```
Comment (rustam-tolipov) [inline person.rb]: **Perfect implementation on your code!** 

- [ ] However, I'm getting errors if I test your app by running `ruby person.rb`. To fix that I suggest you remove all unused super from your code to make your app approvable!

File: person.rb [hunk 2]
```diff
@@ -7,17 +13,52 @@ def initialize(age, name = 'Unknown', parent_permission: true)
...
+end
 
-  public
+class Decorator < Nameable
+  attr_accessor :name
 
-  def can_use_services?
-    of_age? || @parent_permission
+  def initialize(nameable = 'Unknown')
+    super
```
Comment (rustam-tolipov) [inline person.rb]: - [ ] Please accept my suggestion!


```suggestion
```

File: person.rb [hunk 3]
```diff
@@ -1,4 +1,10 @@
-class Person
+class Nameable
```
Comment (rustam-tolipov) [inline person.rb]: - [ ] So, in the previous milestone, you were required to place your classes in a separate file, but your code reviewer approved your code without mentioning that! 

image

In this milestone, I encourage you to place your classes in a separate file. Except for the Person class places other classes into separate `.rb` files. Thank you!

**Review by rustam-tolipov — CHANGES_REQUESTED** [review]
Hello @Mujeeb4582,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
_Your project is almost complete!_
_No Linters errors_

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Check the comments under the review._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb (hunk 1 above)
Comment (Mujeeb4582) [reply person.rb]: @Rustamxon7  when i remove the super then i got linters error

File: person.rb (hunk 2 above)
Comment (Mujeeb4582) [reply person.rb]: @Rustamxon7 what's your suggestion about it

File: person.rb (hunk 3 above)
Comment (Mujeeb4582) [reply person.rb]: @Rustamxon7 In previous Milestone i put everything in separate file but in this milestone Microvese didn't mention it.

**Review by zieeco — APPROVED** [review]
# status: APPROVED :green_circle: :100: 

Hi, @Mujeeb4582, 


68747470733a2f2f6d656469612e67697068792e636f6d2f6d656469612f78543078657a514755357843444a754350652f67697068792e676966


Good job so far 👍 Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Highlights ⚡ 

✅ Descriptive PR 👍 
✅ Good commit messages 👍 
✅ Linters are passing 👏 


## Optional suggestions
- N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**